### PR TITLE
Add DataModule for preprocessed images

### PR DESCRIPTION
**Pull request by TidalPaladin**
To maximize code reuse a DataModule is created to support data preprocessed by torch-dicom. At present, only preprocessed PNG outputs are supported.

### CODE REVIEW

File: torch_dicom/preprocessing/datamodule.py [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+    collate_fn,
+)
+from torch_dicom.datasets.helpers import Transform
+
+
+PathLike = Union[str, os.PathLike, Path]
+
+
+def _prepare_inputs(inputs: Union[PathLike, Sequence[PathLike]]) -> List[Path]:
+    return [Path(i) for i in ([inputs] if isinstance(inputs, (str, os.PathLike, Path)) else inputs)]
```
Comment (timcogan) [inline torch_dicom/preprocessing/datamodule.py:36]: Should be `isinstance(inputs, PathLike)`?

File: torch_dicom/preprocessing/datamodule.py [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+        datasets: Iterable[MetadataDatasetWrapper],
+        mode: Mode,
+    ) -> List[Sampler]:
+        samplers: List[Sampler] = [
+            self.create_sampler((unwrapped := _unwrap_dataset(dataset)), list(unwrapped.files), root, mode)
+            for dataset, root in zip(datasets, roots)
+        ]
+        return samplers
+
+    def setup(self, stage: Optional[str] = None) -> None:
```
Comment (timcogan) [inline torch_dicom/preprocessing/datamodule.py:238]: Consider `StrEnum` for `stage`?

**Review by timcogan — COMMENTED** [review]
Looks good - a couple of minor comments.

File: torch_dicom/preprocessing/datamodule.py (hunk 2 above)
Comment (TidalPaladin) [reply torch_dicom/preprocessing/datamodule.py:238]: This method is a pytorch-lightning hook so we probably need to conform to the signature they provide.

**Review by timcogan — APPROVED** [review]

**Pull request merged**